packages/core/src/modules/routing/RecipientModule.ts

```diff
@@ -178,6 +183,66 @@ export class RecipientModule {
     return event.payload.mediationRecord
   }
 
+  public async provision(mediatorConnInvite: string) {
+    this.logger.debug('Provision Mediation with invitation', { invite: mediatorConnInvite })
+    if (!mediatorConnInvite) {
+      throw new AriesFrameworkError('Mediation provision requires an invitation to be passed')
+    }
+    // Connect to mediator through provided invitation if provided in config
+    // Also requests mediation ans sets as default mediator
+    // Because this requires the connections module, we do this in the agent constructor
+
+    // Assumption: processInvitation is a URL-encoded invitation
+    const invitation = await ConnectionInvitationMessage.fromUrl(mediatorConnInvite)
+    // Check if invitation has been used already
+    if (!invitation || !invitation.recipientKeys || !invitation.recipientKeys[0]) {
+      throw new Error(`Invalid mediation invitation`)
+    }
+    const connection = await this.connectionService.findByInvitationKey(invitation.recipientKeys[0])
+    if (!connection) {
+      this.logger.debug('Mediation Connection does not exist, creating connection')
+      const routing = await this.mediationRecipientService.getRouting()
+
+      const invitationConnectionRecord = await this.connectionService.processInvitation(invitation, {
+        autoAcceptConnection: true,
+        routing: routing,
+      })
+      this.logger.debug('Processed invitation')
+      const { message, connectionRecord: connectionRecord } = await this.connectionService.createRequest(
+        invitationConnectionRecord.id
+      )
+      const outbound = createOutboundMessage(connectionRecord, message)
+      await this.messageSender.sendMessage(outbound)
+
+      // TODO: add timeout to returnWhenIsConnected
+      const completedConnectionRecord = await this.connectionService.returnWhenIsConnected(connectionRecord.id)
+      this.logger.debug('Connection completed, requesting mediation')
+      const mediationRecord = await this.requestAndAwaitGrant(completedConnectionRecord, 60000) // TODO: put timeout as a config parameter
+      this.logger.debug('Mediation Granted, setting as default mediator')
+      await this.setDefaultMediator(mediationRecord)
+      this.logger.debug('Default mediator set')
+      return
+    } else if (connection && !connection.isReady) {
+      const connectionRecord = await this.connectionService.returnWhenIsConnected(connection.id)
```

Comment:
This will probably never resolve. If we need to wait for an action from the other party we're fine. However if we shutdown the agent mid connection and never sent the message, we must manually trigger the next step in the protocol. Not sure what the best approach is here.